civis/tests/test_resources.py

```diff
@@ -68,13 +66,16 @@ def test_create_method_no_iterator_kwarg():
 
     assert 'keyword argument' in str(excinfo.value)
 
+    # Test for TypeError when iterator is present and method takes **kwargs,
+    # but iterator is not valid
```

Comment:
Similarly, maybe this could be
 
> Dynamic functions handle optional argument through a different code path; verify that this also rejects unexpected arguments.